src/lib/Events.ts

```diff
@@ -0,0 +1,18 @@
+import { EventEmitter } from 'events';
+
+import { config } from '../lib/Config';
+import { Logger } from '../lib/Logger';
+import { EventData } from '../types';
+
+export const eventEmitter = new EventEmitter();
+
+export const initializeEventListeners = () => {
+  if (config.eventListeners) {
+    config.eventListeners.forEach(({ event, listener }) => {
+      eventEmitter.addListener(event, (data: EventData) => {
```

Comment:
Could we change the function that's passed into the `.addListener` to be:

```ts
(data?: EventData) => {
  Logger.info(`Emitting event ${event}`, { data });
  listener(data || {});
}
```

That way we can emit events without having to include metadata if there isn't any (e.g. `eventEmitter.emit('STARTUP')`), and it just defaults to an empty object.